### PR TITLE
[DOCS] Fix style inconsistency

### PR DESCRIPTION
**Pull request by yhyakuna**
### Description
What does this PR do?

Going through the front matter and fix inconsistencies:

- `helm` should be `Helm` ([Helm docs)](https://helm.sh/docs/)
- `lambda` should be `Lambda` ([AWS docs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-extensions.html))
- `Vault enterprise` should be `Vault Enterprise`
- `Vault agent` should be `Vault Agent`
- Match the `page_titile` value with the H1 header if they are different
   
   image

### TODO only if you're a HashiCorp employee
- [x] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: docs
-page_title: Vault Secrets Operator Secret Transformation
+page_title: Secret data transformation
 description: >-
   Learn how to transform Secret data with the Vault Secrets Operator.
```
Comment (jonathanfrappier) [inline website/content/docs/platform/k8s/vso/secret-transformation.mdx:5]: secret?
```suggestion
  Learn how to transform secret data with the Vault Secrets Operator.
```

Comment (jonathanfrappier) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx:5]: or is that the formal Kubernetes Secret?

**Review by jonathanfrappier — DISMISSED** [review]

**Review by jonathanfrappier — APPROVED** [review]

File: website/content/docs/platform/k8s/helm/terraform.mdx [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
 page_title: 'Configure Vault Helm using Terraform'
 sidebar_current: 'docs-platform-k8s-terraform'
-description: |-
-  Describes how to configure the Vault Helm chart using Terraform
+description: >-
+  Describes how to configure the Vault Helm chart using Terraform.
 ---
 
-# Configuring Vault helm with terraform
+# Configuring Vault Helm with Terraform
```
Comment (tvoran) [inline website/content/docs/platform/k8s/helm/terraform.mdx:9]: This doesn't match the page_title?

```suggestion
# Configure Vault Helm using Terraform
```

File: website/content/docs/platform/k8s/injector/annotations.mdx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: docs
-page_title: Agent Sidecar Injector Annotations
-description: This section documents the configurable annotations for the Vault Agent Injector.
+page_title: Vault Agent Injector annotations
+description: Learn about the configurable annotations for the Vault Agent Injector.
 ---
 
 # Annotations
```
Comment (tvoran) [inline website/content/docs/platform/k8s/injector/annotations.mdx:7]:
```suggestion
# Vault Agent Injector annotations
```

**Review by tvoran — APPROVED** [review]

File: website/content/docs/platform/k8s/helm/terraform.mdx (hunk 2 above)
Comment (yhyakuna) [reply website/content/docs/platform/k8s/helm/terraform.mdx:9]: I must have missed it somehow... Thank you for catching.

**Review by tvoran — APPROVED** [review]

**Review by jonathanfrappier — APPROVED** [review]

**Pull request merged**